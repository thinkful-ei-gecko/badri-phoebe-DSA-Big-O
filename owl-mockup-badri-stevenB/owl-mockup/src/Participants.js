import React from 'react';



export default function(props){
    const parts = props.xyz.map((val, i) => <li key={i}> <img src={val.avatar} alt={val.name} 
style={{borderRadius: "50%", height: "35px", border: "1px solid black"}}/> 
    {val.name} {val.inSession && <div className="green-dot" />}{!val.inSession && <div className="grey-dot" />}{val.onStage && 'on stage'}</li>)
    return <ul>{ parts }</ul>
}
