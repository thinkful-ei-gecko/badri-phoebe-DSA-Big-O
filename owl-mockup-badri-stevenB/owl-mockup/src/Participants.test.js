import React from 'react';
import ReactDOM from 'react-dom';
import Participants from './Participants';
import renderer from 'react-test-renderer';

describe ("Participants componenet", () => {
    it ("renders without crashing", () => {
        const div = document.createElement('div');
        ReactDOM.render(<Participants xyz={[1,2,3,4]}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it ("renders the UI as expected", () => {
        const tree = renderer
        .create(<Participants xyz={[1,2,3,4]} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})