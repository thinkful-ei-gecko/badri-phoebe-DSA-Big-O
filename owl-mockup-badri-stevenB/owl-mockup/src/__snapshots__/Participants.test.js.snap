// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants componenet renders the UI as expected 1`] = `
<ul>
  <li>
     
    <img
      style={
        Object {
          "border": "1px solid black",
          "borderRadius": "50%",
          "height": "35px",
        }
      }
    />
     
    <div
      className="grey-dot"
    />
  </li>
  <li>
     
    <img
      style={
        Object {
          "border": "1px solid black",
          "borderRadius": "50%",
          "height": "35px",
        }
      }
    />
     
    <div
      className="grey-dot"
    />
  </li>
  <li>
     
    <img
      style={
        Object {
          "border": "1px solid black",
          "borderRadius": "50%",
          "height": "35px",
        }
      }
    />
     
    <div
      className="grey-dot"
    />
  </li>
  <li>
     
    <img
      style={
        Object {
          "border": "1px solid black",
          "borderRadius": "50%",
          "height": "35px",
        }
      }
    />
     
    <div
      className="grey-dot"
    />
  </li>
</ul>
`;
